### ISSUE TITLE
MacOS screen cut off when RetroArch maximized

### ISSUE THREAD

**Issue by matthewbauer** [libretro/RetroArch#6260]
## Description

This has been bugging me recently. I like to use "maximize" window feature that move the entire window to a new desktop (and hides the menu bars). This usually works with most programs but I've had issues doing this with RetroArch.

### Expected behavior

Basically, I want to get the video to display correctly.

### Actual behavior

![screen shot 2018-02-08 at 5 36 23 pm](https://user-images.githubusercontent.com/19036/36004195-f2fb6618-0cf6-11e8-8184-29b0b50cb5a5.png)

Something is wrong with the offset. Note that this does not happen with true RetroArch fullscreen (pressing F) but only with MacOS maximize.

### Steps to reproduce the bug

1. Open RetroArch on MacOS 10.13
2. Click the green maximize button right next to the quit button on the top right of the RetroArch window.
3. Observe cut off video.

### Bisect Results

It has probably always happened and has to do with some Apple API not playing well with RetroArch's video driver. This feature was not available before MacOS 10.10 and I assume it did not have that issue.

### Version/Commit
You can find this information under Information/System Information

- RetroArch: 1.7.0

### Environment information

- OS: macOS 10.13

**Comment by orbea** [comment]
On first impression this looks more like a bug in macOS if their maximize doesn't work, but RetroArch's full screen does.

**Comment by matthewbauer** [comment]
Yes I wonder if MacOS is reporting incorrect values. I don't have the same programs with other program, however.

When you do the "split window" thing it seems to fix itself after you resize. I'll experiment further.

**Comment by orbea** [comment]
Kind of similar to https://github.com/libretro/RetroArch/issues/6284?

**Comment by twinaphex** [comment]
Is this with the OpenGL or Metal driver?

**Comment by CatGreen90** [comment]
FWIW, both RA fullscreen and macOS maximize work fine for me with the Metal driver. I'm pretty sure this issue predates the Metal driver anyway.

**Comment by matthewbauer** [comment]
I can no longer reproduce this. I think it was a bug in macOS High Sierra.

**Comment by gingerbeardman** [comment]
I still have this in High Sierra.

I have to use "Toggle Exclusive Full Screen" to get good full screen display.
